src/timearray.jl

```diff
@@ -331,29 +318,12 @@ lastindex(ta::TimeArray) = length(ta.timestamp)
 
 eachindex(ta::TimeArray) = Base.OneTo(length(ta.timestamp))
 
-# helper methods for inner constructor
-function find_dupes_index(cnames)
-    idx = Int[]
-    for c in 1:length(cnames)
-        if occursin(cnames[c], string(cnames[1:c-1]))
-            push!(idx, c)
-        end
-    end
-    idx
-end
+###### getproperty #################
 
-function replace_dupes(cnames)
-    n = 1
-    while !allunique(cnames)
-        ds = find_dupes_index(cnames)
-        for d in ds
-            if n == 1
-                cnames[d] = string(cnames[d], "_$n")
-            else
-                cnames[d] = string(cnames[d][1:length(cnames[d])-length(string(n))-1], "_$n")
-            end
-        end
-        n += 1
+@generated function getproperty(ta::T, c::Symbol) where {T<:AbstractTimeSeries}
+    fs = fieldnames(T)
+
+    quote
+        (c ∈ $fs) ? getfield(ta, c) : ta[c]
```

Comment:
If a user tries to call a column `values`, for example, this will be incredibly odd behavior. Either replace all field accesses within this package to use `getfield` explicitly, or don't support `getproperty`. (I'd vote for the latter.)

Reply:
Yeah, every use of `ta.values` would need to be changed to `getfield(ta, :values)`.

Reply:
I will change them to our helper function `vaules(ta)`, `timestamp(ta)`, `colnames(ta)` and `meta(ta)`